npm/iptm/iptm.go

```diff
@@ -359,7 +374,9 @@ func (iptMgr *IptablesManager) Add(entry *IptEntry) error {
 
 	log.Logf("Adding iptables entry: %+v.", entry)
 
-	if entry.IsJumpEntry {
+	// Since there is a RETURN statement added to each DROP chain, we need to make sure
+	// any new DROP rule added to ingress or egress DROPS chain is added at the BOTTOM
+	if entry.IsJumpEntry || isDropsChain(entry.Chain) {
```

Comment:
NPM does not need entry.IsJumpEntry anymore with this PR. 
So, please confirm in NPM code.  If we confirm this, we can remove this if condition.
Also, review and update translatePolicy_test.go with this change accordingly. If other codes also impacted, please check their UTs as well.

As I put comment about this before, RETURNs in "AZURE-NPM-EGRESS (or INGRESS)-DROPS" chains  are redundant.

If we confirm this, it seems we only need below block in iptm.go with this PR.
```go
iptMgr.OperationFlag = util.IptablesInsertionFlag.
```

Let me know if I miss something.

Reply:
You are right, we do not need to append when the jumpentry is true, since we are now removing all jump entries and are relying on the default rules to help with that jump transitions to next chains.

As far as the redundant RETURN rule in DROP chains are concerned, i feel strongly to keep those rules in place as a contingency for any later changes. With these redundant rules, we are ensuring that no matter what we will NEVER DROP a MARK'ed packet and this assurance helps are from hitting any other bugs in the future.

Reply:
> One concern in this PR is I am not 100% sure that whether it does allow invalid packets or not allow valid packets when various multiple network policies are applied in AZURE-NPM-INGRESS(or EGRESS)-DROPS chain while I think this PR does not change those behavior from previous code base and getDefaultDropEntries() guarantees that one.
> 
> Do you have any thoughts?

I think overall this is a better design because, keeping the translation logic aside, while evaluating rule we are making sure  that a MARKED packet will never be dropped and a unmarked packet will be dropped on certain conditions. This simplifies  the logic from before where we have multiple avenue to DROPs chain before we evaluated all MARK rules in a given direction (which was wrong).